packages/adblocker/test/engine/engine.test.ts

```diff
@@ -1814,6 +1814,23 @@ foo.com###selector
     );
   });
 
+  describe('#matchCosmeticFilters', () => {
+    it('reports generic hide exceptions', () => {
+      const engine = FilterEngine.parse('###ad\n@@||foo.com^$generichide');
+      const { matches } = engine.matchCosmeticFilters({
+        url: 'https://foo.com',
+        hostname: 'foo.com',
+        domain: 'foo.com',
+        classes: ['ad'],
+        getRulesFromDOM: true,
+      });
+      expect(matches.length).to.be.eql(1);
+      expect(matches[0].exception?.isNetworkFilter()).to.be.true;
```

Comment:
lets have `const match = matches[0]!`; so later we don't need to use `!` or `?`.

Also, lets verify that `filter` is empty

Reply:
addressed in ad999a53beef7f60eb78fa1547bb1d6441455e07